ppg/pg_stat_monitor-autobuild.groovy

```diff
@@ -0,0 +1,382 @@
+    sh """ 
+        set -o xtrace
+        if [ -d aws ]; then
+            rm -rf aws
+        fi
+        if [ ${PLATFORM} = "deb" ]; then
+            sudo apt-get update
+            sudo apt-get -y install wget curl unzip
+        elif [ ${PLATFORM} = "rpm" ]; then
+            sudo yum -y install wget curl unzip
+        fi
+        curl https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip -o awscliv2.zip
+        unzip awscliv2.zip
+        sudo ./aws/install || true
+    """ 
+}
+
+void buildStage(String DOCKER_OS, String STAGE_PARAM) {
+    sh """
+        echo "Docker: $DOCKER_OS, Release: PG$PG_RELEASE, Stage: $STAGE_PARAM"
+        set -o xtrace
+        mkdir -p test
+        wget \$(echo ${GIT_REPO} | sed -re 's|github.com|raw.githubusercontent.com|; s|\\.git\$||')/${BRANCH}/percona-packaging/scripts/pg_stat_monitor_builder.sh -O psm_builder.sh || curl \$(echo ${GIT_REPO} | sed -re 's|github.com|raw.githubusercontent.com|; s|\\.git\$||')/${BRANCH}/percona-packaging/scripts/pg_stat_monitor_builder.sh -o psm_builder.sh
+        if [ -f /etc/redhat-release ]; then
+            sudo yum install -y https://repo.percona.com/yum/percona-release-latest.noarch.rpm || true
+        else
+            DEBIAN_VERSION=\$(lsb_release -sc)
+            sed -e '/.*backports.*/d' /etc/apt/sources.list > sources.list.new
+            sudo mv -vf sources.list.new /etc/apt/sources.list
+            if [ \${DEBIAN_VERSION} = bionic ]; then
+                wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+                echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-11 main" | sudo tee -a /etc/apt/sources.list
+            fi
+            if [ \${DEBIAN_VERSION} = buster ]; then
+                sudo apt-get -y update --allow-releaseinfo-change || true
+            fi
+            if [ \${DEBIAN_VERSION} = stretch ]; then
+                sudo apt-get -y  install apt-transport-https ca-certificates || true
+            fi
+            if [ \${DEBIAN_VERSION} = bullseye ]; then
+                sed -i 's:clang-7::g' psm_builder.sh
+                sed -i '23s:focal:bullseye:' psm_builder.sh
+                sed -i 's:dh-systemd::' psm_builder.sh
+            fi
+            until sudo apt-get update; do
+                sleep 30
+                echo "Waiting ..."
+            done
+            until sudo apt-get -y install gpgv curl clang-11 gnupg; do
+                sleep 30
+                echo "Waiting ..."
+            done
+            sudo wget https://repo.percona.com/apt/percona-release_latest.\$(lsb_release -sc)_all.deb
+            sudo dpkg -i percona-release_latest.\$(lsb_release -sc)_all.deb
+        fi
+
+        sudo percona-release enable ppg-${PG_RELEASE} release
+        pwd -P
+        export build_dir=\$(pwd -P)
+        set -o xtrace
+        cd \${build_dir}
+        if [ -f ./test/pg-stat-monitor.properties ]; then
+            . ./test/pg-stat-monitor.properties
```

Comment:
better to use source command

Reply:
Let's keep it in the current way. **_source_** is a Bash built-in function but we use **_sh_**.